charts/newrelic-k8s-metrics-adapter/tests/deployment_test.yaml

```diff
@@ -0,0 +1,134 @@
+suite: test deployent images and extra
+release:
+  name: my-release
+  namespace: my-namespace
+tests:
+  - it: has the correct image
+    set:
+      personalAPIKey: 21321
+      image:
+        repository: newrelic/newrelic-k8s-metrics-adapter
+        tag: "latest"
+        pullSecrets:
+          - name: regsecret
+      config:
+        accountID: 111
+        region: A-REGION
+    asserts:
+      - matchRegex:
+          path: spec.template.spec.containers[0].image
+          pattern: ^.*newrelic/newrelic-k8s-metrics-adapter:latest
+        template: templates/deployment.yaml
+      - equal:
+          path: spec.template.spec.imagePullSecrets
+          value:
+            - name: regsecret
+        template: templates/deployment.yaml
+  - it: correctly uses the cluster helper
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      cluster: a-cluster
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].env[0].value
+          value: a-cluster
+        template: templates/deployment.yaml
+  - it: renders when extraEnv, extraEnvFrom, extraVolumes and extraVolumeMounts are set
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      extraEnv:
+        - name: ENV_VAR1
+          value: "var1"
+        - name: ENV_VAR2
+          value: "var2"
+      extraEnvFrom:
+        - configMapRef:
+            name: special-config
+      extraVolumes:
+        - name: tmpfs-data
+          emptyDir: {}
+      extraVolumeMounts:
+        - mountPath: /var/db/newrelic-infra/data
+          name: tmpfs-data
+    asserts:
+      - hasDocuments:
+          count: 1
+        template: templates/deployment.yaml
+  - it: renders tolerations
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      tolerations:
+        - key: "key1"
+          operator: "Exists"
+          effect: "NoSchedule"
+    asserts:
+      - equal:
+          path: spec.template.spec.tolerations
+          value:
+            - effect: NoSchedule
+              key: key1
+              operator: Exists
+        template: templates/deployment.yaml
+  - it: renders priorityClassName
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      priorityClassName: high-priority
+    asserts:
+      - equal:
+          path: spec.template.spec.priorityClassName
+          value: high-priority
+        template: templates/deployment.yaml
+  - it: renders nodeSelector
+    set:
+      personalAPIKey: 21321
+      config:
+        accountID: 111
+        region: A-REGION
+      nodeSelector:
+        disktype: ssd
+    asserts:
+      - equal:
+          path: spec.template.spec.nodeSelector
+          value:
+            disktype: ssd
+        template: templates/deployment.yaml
+  - it: renders affinity
```

Comment:
Affinity comes [from the common-library](https://github.com/newrelic/helm-charts/blob/master/library/common-library/templates/_affinity.tpl) and [it is tested](https://github.com/newrelic/helm-charts/blob/master/library/CHART-TEMPLATE/tests/common_library_affinity_test.yaml). Is there any reason to re-test it?

Reply:
It's to test they get rendered, if the template has a problem the library won't render them, whenever there is an if, I like to put a test with the happy path.

Reply:
same as the other, only added to one test with all the values ensuring it gets rendered.